File: frontend-app/src/core/model/dto/response/EmpresaResponse.ts

export interface EmpresaResponse {
    id : number;
    nome : string;
    emailNotif : string;
    telegramChatId : string;
    porcentagemMaxFalhasPorLote : number;    
    maxDispositivosQuant : number;
    minTempoParaProxNotif : number;
    diaPagto : number;
    temporario : boolean;
    usoTemporarioPor : number;
    bloqueada : boolean;
    criadoEm : Date;
    usoRegularIniciadoEm : Date;
    pagoAte : Date
}

export const DEFAULT_EMPRESA_OBJ : EmpresaResponse = {
    id: 0,
    nome: '',
    emailNotif: '',
    telegramChatId: '',
    porcentagemMaxFalhasPorLote: 0,
    maxDispositivosQuant: 0,
    minTempoParaProxNotif: 0,
    diaPagto: 0,
    temporario: false,
    usoTemporarioPor: 0,
    bloqueada: false,
    criadoEm: new Date( 0 ),
    usoRegularIniciadoEm: new Date( 0 ),
    pagoAte: new Date( 0 )
};

File: main-api/src/main/java/com/italomonitor/main/dto/response/EmpresaResponse.java
package com.italomonitor.main.dto.response;

import java.util.Date;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@Builder
public class EmpresaResponse {

    private Long id;
    
    private String nome;
    
    private String emailNotif;
    
    private String telegramChatId;
    
    private double porcentagemMaxFalhasPorLote;
    
    private int maxDispositivosQuant;
    
    private int minTempoParaProxNotif;
    
    private int diaPagto;
    
    private boolean temporario;
    
    private int usoTemporarioPor;
    
    private boolean bloqueada;
    
    private Date criadoEm;
    
    private Date usoRegularIniciadoEm;
    
    private Date pagoAte;

}
